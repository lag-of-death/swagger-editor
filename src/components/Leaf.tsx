import * as React from "react";

export interface ILeafProps {
  renderIssuesIndicators: () => React.FunctionComponentElement<{}>;
  labelKey: string;
}

const Leaf = ({labelKey, renderIssuesIndicators}: ILeafProps) => {
  return (
    <li>
      {renderIssuesIndicators()}
      {labelKey}
    </li>
  );
};

export default Leaf;
