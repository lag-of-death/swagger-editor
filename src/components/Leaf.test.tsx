import { shallow } from "enzyme";
import * as React from "react";
import Leaf from "./Leaf";

describe("Leaf", () => {
  test("rendering", () => {
    const component = shallow(<Leaf labelKey="testLabelKey" renderIssuesIndicators={() => <div></div>}/>);

    expect(component).toMatchSnapshot();
  });
});
