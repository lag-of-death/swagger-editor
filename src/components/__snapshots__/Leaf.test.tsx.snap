// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaf rendering 1`] = `
<styled.li>
  testLabelKey
  <styled.div>
    <div />
  </styled.div>
</styled.li>
`;
